packages/ember-metal-views/lib/renderer.js

```diff
@@ -170,9 +176,17 @@ Renderer.prototype.willRender = function (view) {
 
 Renderer.prototype.remove = function (view, shouldDestroy) {
   this.willDestroyElement(view);
-  view._transitionTo('destroying', false);
+  run.scheduleOnce('render', this, this.renderElementRemoval, view);
```

Comment:
Can you rewrite the logic to avoid using `scheduleOnce`? You can allow renderElementRemoval to be called multiple times, but just no-op if the action has already been performed (using flags on the view).